File: Builder/python-source/builders/VegiburgerBuilder.py
from base.Burger import Burger
from builders.BurgerBuilder import BurgerBuilder


class VegiburgerBuilder(BurgerBuilder):
    def __init__(self):
        super().__init__()
        self._contents = []

    def reset(self):
        self._contents = []

    def add_low_bread(self):
        self._contents.append('GrayLoaf')

    def add_meat(self):
        pass

    def add_high_bread(self):
        self._contents.append('BlackLoaf')

    def add_vegetable(self):
        self._contents.append('Salade')

    def get_burger(self) -> Burger:
        return Burger(self._contents)

File: JavaSource/src/main/java/ru/akhcheck/patterns/builder/builders/VegiburgerBuilder.java
package ru.akhcheck.patterns.builder.builders;

import ru.akhcheck.patterns.builder.base.Burger;

import java.util.LinkedList;
import java.util.List;

public class VegiburgerBuilder implements BurgerBuilder {
    private List<String> components = new LinkedList<>();

    @Override
    public BurgerBuilder reset() {
        components.clear();
        return this;
    }

    @Override
    public BurgerBuilder addLowBread() {
        components.add("GrayLoaf");
        return this;
    }

    @Override
    public BurgerBuilder addMeat() {
        return this;
    }

    @Override
    public BurgerBuilder addHighBread() {
        components.add("BlackLoaf");
        return this;
    }

    @Override
    public BurgerBuilder addVegetable() {
        components.add("Salade");
        return this;
    }

    @Override
    public Burger getBurger() {
        return new Burger(components);
    }
}
